### ISSUE TITLE
please check your internet message

### ISSUE THREAD

**Issue by BurierAbdalwahhabElia** [HekaHealth/heka-health-plugin#2]
my internet connection is very good.
when i press (connect) button. this message always appear (please check your internet!).
why this appear? i don't think is about internet connection, may be is a bug or error in plugin.
i need help as soon as possible, please.

**Comment by Pulkit07** [comment]
Hey @BurierAbdalwahhabElia, thanks for reporting.

Which platform are you trying to connect? Google Fit or Apple Healthkit?

In the case of Google Fit, have you set up a Google Client ID and the redirect URIs?

1. You can get Google Client id by following the link: https://developers.google.com/fit/android/get-api-key
2. To setup redirect URI, you need to make following changes to `android/app/src/main/AndroidManifest.xml`:

```
...
    <application
        ...
        ...
        # Add the line below
        xmlns:tools="http://schemas.android.com/tools"
    >
        ...
        ...
        # Add this whole activity
        <activity
          android:name="net.openid.appauth.RedirectUriReceiverActivity"
          android:exported="true"
          tools:node="replace">
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data
                  android:scheme="com.googleusercontent.apps.<YOUR_GOOGLE_CLIENT_ID>"
                />
            </intent-filter>
        </activity>
    </application>
```

If this doesn't work, you can also email us at `contact at hekahealth dot co` and we can help you over a call.

**Comment by BurierAbdalwahhabElia** [comment]
Hi, thank you for response.
My platform is google fit.
I already done with google client id and redirect url as you mention literally.
But i want to ask you about this line of code:
  static const _apiKey = '7368bad8-aadd-4624-a58c-7e8af2b3cfb7';
Is this my generated key of my app in heka dashboard? If yes, I set this right also.
Have I fill fields of privacy url,term url, domains and home app page url in settings of my project in google console?
And I want to ask about have I install google fit application in my device to running as expected without errors?

**Comment by Pulkit07** [comment]
> Hi, thank you for response. My platform is google fit. I already done with google client id and redirect url as you mention literally.

Okay, looking at our server logs, it seems like you missed adding the Google Client Id in the Heka dashboard. You can add that using the following steps:

1) Go to [Heka Dashboard](https://appdev.hekahealth.co)
2) Click on app that you have created, an edit app view will open
3) Enter the Google Auth Client ID you received from Google there in the last text field
4) Click on Submit to save

> But i want to ask you about this line of code:   static const _apiKey = '7368bad8-aadd-4624-a58c-7e8af2b3cfb7'; Is this my generated key of my app in heka dashboard?

Yes, it is the generated key of your app from the Heka dashboard.

**Comment by BurierAbdalwahhabElia** [comment]
Hi, I already add google client id in heka dashboard when I created first time. But in all times I edit project in heka dashboard I found field of google client id empty. Is this normal?

**Comment by Pulkit07** [comment]
> Hi, I already add google client id in heka dashboard when I created first time. But in all times I edit project in heka dashboard I found field of google client id empty. Is this normal?

Yes, it's a bug. Apologies about that. Our team is working on fixing the underlying root cause, and should be fixed soon.

Meanwhile, to get things started, you can go to edit view, enter the Google Auth Client Id and hit `Submit`. This shall update the client id on the server and should get you good to go for testing.

I will also make sure to update once the dashboard bug is fixed.

**Comment by Pulkit07** [comment]
Hey @BurierAbdalwahhabElia, the dashboard bug is fixed. Thanks for pointing us to it. Let us know if you still face issues connecting to `Google Fit`.

**Comment by Pulkit07** [comment]
Hey @BurierAbdalwahhabElia, we have released a new version of package `0.0.17`. It fixes the error message by mentioning about missing client id. We are closing this issue for now.

**Comment by BurierAbdalwahhabElia** [comment]
Hi,
I am very appreciated your work to solve this bug and release a new version, my issue in connect of heka wealth widget is solved and every think is ok.
Thank you.
